### DISCUSSION TITLE
Edge Classification without node features

### DISCUSSION THREAD

**Discussion by ntquanghai** [pyg-team/pytorch_geometric#7315] [Q&A]
Hi, I am currently working on a transaction graph dataset, where each transaction is an edge between two nodes. The nodes do not include any features, while the edges will contain the majority of the transaction details. Moreover, between two nodes, there can exist many edges, as in there can be many transactions between two people. My goal is to detect whether a transaction is fraudulent or not, and so I believe this to be an Edge Classification problem. 
So far, I have opted to build a model which will treat edges the same way as nodes, and replace the node features with edge features in the parameters. Moreover, I have also seen some similar examples, which include the edge features in the edge_attr of some operators, then generalize a tensor of zeros as node features, then use them as inputs for the models
So I have some questions

1.  Is my original solution valid, or would it break the architecture? If possible, I hope you can suggest an alternative approach to my problem
2.  Regarding the examples, would this return the node embeddings, or edge embeddings?

I apologize if the questions seem a bit simple, I am a beginner to the field and some concepts are still a bit hard for me to grasp. I would appreciate any response, and thank you all very much!

**Comment by rusty1s** [comment]
1. I actually think your solution is superior here as you will actually learn a representation per transaction, rather than just learning representations per user.
2. In your case, you would return the node embeddings. If you want to frame this as a edge classification problem, you would combine source and destination user embeddings and edge features for the final classification, e.g.:
```python
MLP(torch.cat([src_user, dst_user, edge_attr], dim=-1))

**Reply by ntquanghai** [reply]
That clears up a lot, thanks for the suggestion!
